src/docs/codelabs/layout-basics_migrated.md

```diff
@@ -0,0 +1,3340 @@
+  final innerColumnWidget = innerColumnElement.widget as Column;
+
+  if (innerColumnWidget.crossAxisAlignment != CrossAxisAlignment.start) {
+    _result(false, ['The Column that contains the name and title should use CrossAxisAlignment.start as its CrossAxisAlignment value.']);
+    return;
+  }
+
+  if (innerColumnWidget.mainAxisSize != MainAxisSize.min) {
+    _result(false, ['The Column that contains the name and title should use MainAxisSize.min as its MainAxisSize value.']);
+    return;
+  }
+
+  // Check inner Column has two Text children
+
+  if (innerColumnWidget.children.any((w) => w is! Text)) {
+    _result(false, ['The Column that contains the name and title should have two children, both Text widgets.']);
+    return;
+  }
+
+  // Check first Text has headline style
+
+  final nameText = innerColumnWidget.children[0] as Text;
+
+  if (nameText.style?.fontSize != 24) {
+    _result(false, ['The Text widget for the name should use the "headline5" textStyle.']);
+    return;
+  }
+
+  _result(true);
+}
+{$ end test.dart $}
+```
+
+#### Exercise: Wrap the Column in a Row
+{:.no_toc}
+{{site.alert.secondary}}
+
+  Wrap the `Column` you implemented in a
+  `Row` that contains the following widgets:
+
+<ul markdown="1">
+  <li markdown="1">
+  An `Icon` widget set to `Icons.account_circle`
+  and with a size of 50 pixels.
+
+  </li>
+  <li markdown="1">
+  A `Padding` widget that creates a space of 8
+  pixels around the `Icon` widget.
+
+  To do this, you can specify `const EdgeInsets.all(8.0)`
+  for the `padding` property.
+
+  The `Row` should look like this:
+  </li>
+</ul>
+
+  ```dart
+     Row(
+       children: [
+        Padding(
+          padding: const EdgeInsets.all(8.0),
+          child: Icon(Icons.account_circle, size: 50),
+        ),
+        Column( ... ), // <--- The Column you first implemented
+      ],
+     );
+  ```
+
+```run-dartpad:theme-dark:mode-flutter:run-true:width-100%:height-400px:split-60:null_safety-true
+{$ begin main.dart $}
+import 'dart:async';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+class MyWidget extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      mainAxisSize: MainAxisSize.min,
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(
+          'Flutter McFlutter',
+          style: Theme.of(context).textTheme.headline5,
+        ),
+        Text(
+          'Experienced App Developer',
+        ),
+      ],
+    );
+  }
+}
+{$ end main.dart $}
+{$ begin solution.dart $}
+import 'dart:async';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+class MyWidget extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        Padding(
+            padding: const EdgeInsets.all(8.0),
+            child: Icon(Icons.account_circle, size: 50)),
```

Comment:
In general, we always try to use trailing commas to keep things indented by two characters, in part because DartPad's width is so constrained when shown in an iframe.

Here, if there was a comma in between the two parentheses at the end of the line, the last one would be moved to the next line, and both parameters for `Padding` would be indented two character instead of four.

Reply:
Agreed, I think this was present in the original Gist, I will fix it now before merge.